universal-application-tool-0.0.1/app/assets/javascripts/azure_delete.ts

```diff
@@ -0,0 +1,27 @@
+/**
+ * This is responsible for deleting a file that was uploaded to Azure blob storage.
+ */
+
+class AzureDeleteController {
+  constructor() {
+    const deleteContainer = document.getElementById('fileupload-delete-button')
+    deleteContainer.addEventListener('click', (event) => this.attemptDelete())
+  }
+
+  attemptDelete() {
+    const azblob = window['azblob']
+    const blockBlobUrl = this.getBlockBlobUrl()
+    if (!blockBlobUrl) {
+      return
+    }
+    blockBlobUrl.delete(azblob.Aborter.none)
+  }
+
+  private getBlockBlobUrl() {
+    const azblob = window['azblob']
+    const searchParams = new URLSearchParams(document.location.search)
+    return new azblob.BlockBlobURL(searchParams.get('blockBlobUrlString'))
```

Comment:
how does this work if the blockBlobUrlString is empty?

Reply:
I updated this now, if blockBlobURLString doesn't exist as part of the URL we'll get a null result and return null and no delete action will be taken. Let me know how this looks